specs/enhance_gluster_cluster_sync_logic.adoc

```diff
@@ -0,0 +1,275 @@
+= Enhance gluster cluster synchronization logic as per recommendations
+
+The intent of this change is to accommodate additional details available as part
+of `get-state` output while syncing the gluster cluster details and store them
+in central store.
+
+Also instead of using `gstatus` for cluster health and volume utilization
+details, we need to user recommendations provided by gluster team.
+
+
+== Problem description
+
+This specification talks about various changes needed to be done under gluster
+cluster synchronization code to accomodate the additional details related to
+volume snapshots, geo-rep sessions.
+
+Also as `gtstaus` is quite a unstable and also few scenarios are not taken care
+of, using it for deciding the overall cluster status and volume utilization is
+not appropriate. Rather we need to write the aggregations logic to decide the
+cluster state based on peers status and volume status of the gluster trusted
+storage pool. Also the overall volume utilization should be calculated based on
+aggregations done for the participating bricks of the volumes.
+
+
+== Use Cases
+
+This addresses the synchronization of gluster cluster details like
+
+* Volume snapshot details
+* Add volume re-balance estimation time details
+* Volume geo-rep details
+* Client connections per volume
+* Identification of arbiter volumes in cluster (if any)
+* Instead of `gstatus` use aggregations across bricks utilization to calculate
+volume utilization
+* Instead of `gstatus` use peer status and volume status to aggregate and
+calculate the cluster status
+
+
+== Proposed change
+
+* Use `get-state` output to figure out the volume snapshot details and update in
+central store
+
+* Figure out the re-balance estimation time from `get-state` output and update
+for volume in central store
+
+* Use `get-state` output for geo-rep session details like no. of geo-rep
+sessions and active geo-rep sessions and update per volume in central store
+
+* Figure out the client connections per brick from `get-state` output and
+aggregate for volume and update in central store
+
+* Identify a arbiter volume and mark accordingly in central store, based on
```

Comment:
In `get-state` output we do get the `arbiter_count = 1` for an arbiter volume but there is no marking of the arbiter brick as such. A sample output is as below

```
[Volumes]
Volume1.name: arb-vol
Volume1.id: 543fd98b-b876-4027-826a-c051761b5b40
Volume1.type: Replicate
Volume1.transport_type: tcp
Volume1.status: Started
Volume1.brickcount: 3
Volume1.Brick1.path: dhcp43-118.lab.eng.blr.redhat.com:/root/gluster_bricks/vol3_b1
Volume1.Brick1.hostname: dhcp43-118.lab.eng.blr.redhat.com
Volume1.Brick2.path: dhcp43-116.lab.eng.blr.redhat.com:/root/gluster_bricks/vol3_b2
Volume1.Brick2.hostname: dhcp43-116.lab.eng.blr.redhat.com
Volume1.Brick3.path: dhcp43-126.lab.eng.blr.redhat.com:/root/gluster_bricks/vol3_b3
Volume1.Brick3.hostname: dhcp43-126.lab.eng.blr.redhat.com
Volume1.Brick3.port: 49152
Volume1.Brick3.rdma_port: 0
Volume1.Brick3.status: Started
Volume1.Brick3.signedin: True
Volume1.snap_count: 0
Volume1.stripe_count: 1
Volume1.replica_count: 3
Volume1.subvol_count: 1
Volume1.arbiter_count: 1
Volume1.disperse_count: 0
Volume1.redundancy_count: 0
Volume1.quorum_status: not_applicable
Volume1.snapd_svc.online_status: Offline
Volume1.snapd_svc.inited: True
Volume1.rebalance.id: 00000000-0000-0000-0000-000000000000
Volume1.rebalance.status: not_started
Volume1.rebalance.failures: 0
Volume1.rebalance.skipped: 0
Volume1.rebalance.lookedup: 0
Volume1.rebalance.files: 0
Volume1.rebalance.data: 0Bytes
Volume1.options.transport.address-family: inet
Volume1.options.performance.readdir-ahead: on
Volume1.options.nfs.disable: on
```

@samikshan can we add that detail here?